src/components/pages/room.js

```diff
@@ -9,33 +9,39 @@ import { useUser } from '../user-context.js';
 import FavoriteStar from '../favorite-star.js';
 import DeleteButton from '../delete-button.js';
 import { useTranslation } from 'react-i18next';
-import { MailOutlined } from '@ant-design/icons';
 import { useDateFormat } from '../locale-context.js';
 import RoomMetadataForm from '../room-metadata-form.js';
 import DeleteIcon from '../icons/general/delete-icon.js';
 import { handleApiError } from '../../ui/error-helper.js';
 import MoveUpIcon from '../icons/general/move-up-icon.js';
 import MoveDownIcon from '../icons/general/move-down-icon.js';
-import SettingsIcon from '../icons/main-menu/settings-icon.js';
+import { DragOutlined, MailOutlined } from '@ant-design/icons';
 import DuplicateIcon from '../icons/general/duplicate-icon.js';
+import DragAndDropContainer from '../drag-and-drop-container.js';
 import RoomApiClient from '../../api-clients/room-api-client.js';
 import { useSessionAwareApiClient } from '../../ui/api-helper.js';
 import DocumentMetadataModal from '../document-metadata-modal.js';
-import React, { Fragment, useEffect, useRef, useState } from 'react';
+import { Button, Tabs, message, Tooltip, Breadcrumb } from 'antd';
 import DocumentApiClient from '../../api-clients/document-api-client.js';
 import RoomExitedIcon from '../icons/user-activities/room-exited-icon.js';
-import { ensureIsExcluded, swapItemsAt } from '../../utils/array-utils.js';
 import IrreversibleActionsSection from '../irreversible-actions-section.js';
-import { Button, Tabs, message, Tooltip, Breadcrumb, Dropdown } from 'antd';
 import RoomInvitationCreationModal from '../room-invitation-creation-modal.js';
 import { FAVORITE_TYPE, DOC_VIEW_QUERY_PARAM } from '../../domain/constants.js';
+import { isRoomInvitedCollaborator, isRoomOwner } from '../../utils/room-utils.js';
 import { DOCUMENT_METADATA_MODAL_MODE } from '../document-metadata-modal-utils.js';
-import { isRoomOwner, isRoomOwnerOrInvitedCollaborator } from '../../utils/room-utils.js';
+import { ensureIsExcluded, moveItem, swapItemsAt } from '../../utils/array-utils.js';
+import React, { Fragment, useEffect, useId, useMemo, useRef, useState } from 'react';
 import { roomShape, invitationShape, documentExtendedMetadataShape } from '../../ui/default-prop-types.js';
 import { confirmDocumentDelete, confirmRoomDelete, confirmRoomMemberDelete, confirmRoomInvitationDelete, confirmLeaveRoom } from '../confirmation-dialogs.js';
 
 const logger = new Logger(import.meta.url);
 
+const VIEW_MODE = {
+  owner: 'owner',
+  collaborator: 'collaborator',
+  member: 'member'
```

Comment:
Just a thought: what about `nonCollaboratingMember`?

Reply:
Makes sense, changed to `collaboratingMember` and `nonCollaboratingMember`.